modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/segment_edit/SegmentEdit.es.js

```diff
@@ -239,6 +242,37 @@ class SegmentEdit extends Component {
 		) : null;
 	};
 
+	/**
+	 * Decides wether to redirect the user or warn
+	 *
+	 * @memberof SegmentEdit
+	 */
+	_handleCancelButton = () => {
+		const {hasChanged} = this.state;
+
+		if (hasChanged) {
+			const confirmed = confirm(
+				Liferay.Language.get('criteria-cancel-confirmation-message')
+			);
+			if (confirmed) {
```

Comment:
new line above this to separate non-sortable code. If `confirmed` was below the `if` block, it'd break.